src/client-proxy.ts

```diff
@@ -0,0 +1,65 @@
+import { ClientProxy } from "@nestjs/microservices";
+import axios from "axios";
+import { resolve } from "dns";
+
+export class JSONRPCClient extends ClientProxy {
+  constructor(private readonly url: string) {
+    super();
+  }
+  private counter: number = 0;
+  private jsonrpc: string = "2.0";
+
+  connect(): Promise<any> {
+    throw new Error('The "connect()" method is not supported in JSONRPC mode.');
+  }
+  close() {
+    return Promise.resolve();
+  }
+  /**
+   * Method is unsupported for JSONRPC
+   */
+  protected publish(packet: any, callback: (packet: any) => any): any {
+    throw new Error("Method is not supported in JSONRPC mode.");
+  }
+  /**
+   * Method is unsupported for JSONRPC
+   */
+  protected async dispatchEvent(packet: any): Promise<any> {
+    throw new Error("Method is not supported in JSONRPC mode.");
+  }
+  getService<SvcInterface>(namespace: string): ServiceClient<SvcInterface> {
+    let url = this.url;
+    let id = this.counter;
+    let jsonrpc = this.jsonrpc;
+    return new Proxy(
+      {},
+      {
+        get(obj, prop) {
+          return function(params: any) {
+            return axios
+              .post(url, {
+                method: namespace + "." + prop.toString(),
+                params,
+                jsonrpc: "2.0",
+                id: ++id
+              })
+              .then(res => ({ jsonrpc, result: res, id }))
+              .catch(err => {
+                const { code, message, data } = err.response.data;
+                let resp = { code, message, data };
+
+                return { jsonrpc, error: resp, id };
```

Comment:
It should recreate the CodedRpcException and throw it here.

Reply:
cc @alantreadway-hfour whether we want to deal with different errors in a different way here.

One option is to just rethrow any error type. The other is to differentiate between failures and successes.

I personally think rethrowing is just fine for the basic client, then other clients can be built on top of it that catch errors and decide what to do with them based on the error code and error message (maybe retry if idempotent request and if the code indicates a retry might be possible)

Reply:
Let's start simple, we can add more advanced (retry) behaviours later? So rethrow seems fine here to me.